tests/DeprecationTest.php

```diff
@@ -222,16 +212,20 @@ public function testDeprecationWithIgnoredLink(): void
 
     public function testDeprecationIfCalledFromOutside(): void
     {
-        Deprecation::enableWithTriggerError();
+            Deprecation::enableWithTriggerError();
 
-        $this->expectErrorHandler(
-            'Bar::oldFunc() is deprecated, use Bar::newFunc() instead. (Bar.php:%d called by Foo.php:14, https://github.com/doctrine/foo, package doctrine/foo)',
-            'https://github.com/doctrine/foo'
-        );
+        try {
+            $this->expectErrorHandler(
+                'Bar::oldFunc() is deprecated, use Bar::newFunc() instead. (Bar.php:%d called by Foo.php:14, https://github.com/doctrine/foo, package doctrine/foo)',
+                'https://github.com/doctrine/foo'
+            );
 
-        Foo::triggerDependencyWithDeprecation();
+            Foo::triggerDependencyWithDeprecation();
 
-        $this->assertEquals(1, Deprecation::getUniqueTriggeredDeprecationsCount());
+            $this->assertEquals(1, Deprecation::getUniqueTriggeredDeprecationsCount());
+        } finally {
+            restore_error_handler();
```

Comment:
Can't `tearDown()` be used in this test class just like in `tests/VerifyDeprecationsTest.php`?

Reply:
It could, but this is safer: if for some reason something fails before we get to set the error handler, it's better not to restore it.